integration-tests/actions/vrf/vrfv2plus/setup_steps.go

```
@@ -235,51 +232,100 @@ func setupVRFNode(contracts *vrfcommon.VRFContracts, chainID *big.Int, config *v
 	nodeConfig := node.NewConfig(vrfNode.CLNode.NodeConfig,
 		node.WithKeySpecificMaxGasPrice(vrfNode.TXKeyAddressStrings, *config.CLNodeMaxGasPriceGWei),
 	)
-	l.Info().Msg("Restarting Node with new sending key PriceMax configuration")
+	l.Info().
+		Strs("Sending Keys", vrfNode.TXKeyAddressStrings).
+		Int64("Price Max Setting", *config.CLNodeMaxGasPriceGWei).
+		Msg("Restarting Node with new sending key PriceMax configuration")
 	err = vrfNode.CLNode.Restart(nodeConfig)
 	if err != nil {
 		return fmt.Errorf(vrfcommon.ErrGenericFormat, vrfcommon.ErrRestartCLNode, err)
 	}
 	return nil
 }
 
-func SetupVRFV2PlusWrapperEnvironment(
+func SetupVRFV2PlusWrapperForExistingEnv(
 	ctx context.Context,
+	sethClient *seth.Client,
+	vrfContracts *vrfcommon.VRFContracts,
+	vrfv2PlusTestConfig types.VRFv2PlusTestConfig,
+	consumerContractsAmount int,
 	l zerolog.Logger,
+) (*VRFV2PlusWrapperContracts, *big.Int, error) {
+	config := *vrfv2PlusTestConfig.GetVRFv2PlusConfig()
+	var wrapper contracts.VRFV2PlusWrapper
+	var err error
+	if *config.ExistingEnvConfig.UseExistingWrapper {
+		wrapper, err = contracts.LoadVRFV2PlusWrapper(sethClient, *config.ExistingEnvConfig.WrapperAddress)
+		if err != nil {
+			return nil, nil, fmt.Errorf(vrfcommon.ErrGenericFormat, "error loading VRFV2PlusWrapper", err)
+		}
+	} else {
```

I think we should throw an error here since the function is already called `SetupVRFV2PlusWrapperForExistingEnv`. You're double checking if `useExistingWrapper` in this function and the caller function.

---

@leeyikjiun In the caller function, Im calling `*config.VRFv2Plus.General.UseExistingEnv`, which is not the same as `useExistingWrapper`


so there might be cases when
1. we need to use existing env, meaning we need to reuse existing Coordinator contract.
- if thats the case, then we also might deploy our own VRFWrapper contract - like in Nexon chain, where we dont have one.
- or use already deployed VRFWrapper contract

have I understood your comment correctly?

---

Oh, my bad. I thought they were the same flag. I still don't like the design here though, as explained by the other comment. You can resolve this conversation since the original comment is no longer relevant.